pkgs/by-name/vp/vpp/package.nix

```diff
@@ -0,0 +1,141 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  nix-update-script,
+  cmake,
+  pkg-config,
+  check,
+  openssl,
+  python3,
+  subunit,
+  mbedtls_2,
+  libpcap,
+  libnl,
+  libmnl,
+  libelf,
+  dpdk,
+  jansson,
+  zlib,
+  rdma-core,
+  libbpf,
+  xdp-tools,
+  enableDpdk ? true,
+  enableRdma ? true,
+  # FIXME: broken: af_xdp plugins - no working libbpf found - af_xdp plugin disabled
+  enableAfXdp ? false,
+}:
+
+let
+  dpdk' = dpdk.overrideAttrs (old: {
+    mesonFlags = old.mesonFlags ++ [ "-Denable_driver_sdk=true" ];
+  });
+
+  rdma-core' = rdma-core.overrideAttrs (old: {
+    cmakeFlags = old.cmakeFlags ++ [
+      "-DENABLE_STATIC=1"
```

Comment:
It does not work with shared libs?

Reply:
The RDMA plugin is [hardcoded](https://github.com/FDio/vpp/blob/a7e01a83beadc70d94be64c89cf091aaa941d3d8/src/plugins/rdma/CMakeLists.txt#L21-L22) to use static libs, it might change in the future but currently it's necessary to link it statically.